### PR TITLE
Ability to have sort col by view

### PR DESCRIPTION
**Pull request by vuon9**
Support separating sort constraint for different views.

### CODE REVIEW

File: cointop/cointop.go [hunk 1]
```diff
@@ -284,9 +288,12 @@ func NewCointop(config *Config) (*Cointop, error) {
...
 			shortcutKeys:          DefaultShortcuts(),
-			sortBy:                DefaultSortBy,
-			page:                  0,
-			perPage:               int(perPage),
+			selectedView:          CoinsView,
+			viewSorts: map[string]*sortConstraint{
+				CoinsView: {DefaultSortBy, false},
+			},
+			page:    0,
+			perPage: int(perPage),
```
Comment (lyricnz) [inline cointop/cointop.go:296]: what's going on with the indent here?

File: cointop/table.go [hunk 2]
```diff
@@ -274,6 +268,11 @@ func (ct *Cointop) ToggleTableFullscreen() error {
 func (ct *Cointop) SetSelectedView(viewName string) {
 	ct.State.lastSelectedView = ct.State.selectedView
 	ct.State.selectedView = viewName
+
+	// init sort constraint for the view in case user travels it in the first time
```
Comment (lyricnz) [inline cointop/table.go:272]: "init sort constraint for the view if it hasn't been seen before" ?

File: cointop/table_header.go [hunk 3]
```diff
@@ -217,12 +217,12 @@ func (ct *Cointop) UpdateTableHeader() error {
 		arrow := " "
 		colorfn := baseColor
 		if !noSort {
-			if ct.State.sortBy == col {
+			corSortCons := ct.State.viewSorts[ct.State.selectedView]
```
Comment (lyricnz) [inline cointop/table_header.go:220]: do you mean cur (current)?

File: cointop/sort.go [hunk 4]
```diff
@@ -89,15 +89,15 @@ func (ct *Cointop) Sort(sortBy string, desc bool, list []*Coin, renderHeaders bo
 // SortAsc sorts list of coins in ascending order
 func (ct *Cointop) SortAsc() error {
 	log.Debug("SortAsc()")
-	ct.State.sortDesc = false
+	ct.State.viewSorts[ct.State.selectedView].sortDesc = true
```
Comment (miguelmota) [inline cointop/sort.go:92]: I think this needs to be `false`?

**Comment by miguelmota** [comment]
Looks good! Can be merged after all comments have been address @vuon9

File: cointop/sort.go (hunk 4 above)
Comment (vuon9) [reply cointop/sort.go]: Yes! My mistake!

File: cointop/table_header.go (hunk 3 above)
Comment (vuon9) [reply cointop/table_header.go]: Yes, "current" is true, I've changed the var name.

File: cointop/table.go (hunk 2 above)
Comment (vuon9) [reply cointop/table.go]: You're right! I've changed as the suggestion.

File: cointop/cointop.go (hunk 1 above)
Comment (vuon9) [reply cointop/cointop.go]: That's the normal behavior of lint tool, I can't answer why 😂

**Pull request merged**